src/raglite/_query_adapter.py

```diff
@@ -157,6 +158,7 @@ def update_query_adapter(  # noqa: PLR0915, C901
             raise ValueError(error_message)
         # Store the optimal query adapter in the database.
         index_metadata = session.get(IndexMetadata, "default") or IndexMetadata(id="default")
-        index_metadata.metadata_ = {**index_metadata.metadata_, "query_adapter": A_star}
+        index_metadata.metadata_["query_adapter"] = A_star
+        flag_modified(index_metadata, "metadata_")
```

Comment:
It seems I had mistakenly assumed that recreating the dict would mark it as dirty. This seems better as it's more explicit!

Reply:
No, I think you are right that it is flagged as dirty, since you remake it. However, when something is flagged as dirty, SQLAlchemy checks if anything changed. (i think)

That causes it to perform some check like 'x == y', and because numpy arrays cannot be compared like that it crashes. Flagging modified skips this comparison.

Reply:
TIL, thanks!